INFERENCE.md

```diff
@@ -0,0 +1,202 @@
+This method of playing incremental generations reduces the latency of the Parler-TTS model from the total time to generate 1720 tokens, to the time taken to play the first chunk of audio (86 tokens). This can result in significant improvements to perceived latency,  particularly when the chunk size is chosen to be small. In practice, the chunk size should be tuned to your device: using a smaller chunk size will mean that the first chunk is ready faster, but should not be chosen so small that the model generates slower than the time it takes to play the audio.
+
+
+### How Can I Use It?
+
+We've added [ParlerTTSStreamer](https://github.com/huggingface/parler-tts/blob/main/parler_tts/streamer.py) to the library. Don't hesitate to adapt it to your use-case.
+
+Here's how to create a generator out of the streamer.
+
+```py
+import torch
+from parler_tts import ParlerTTSForConditionalGeneration, ParlerTTSStreamer
+from transformers import AutoTokenizer
+from threading import Thread
+
+torch_device = "cuda:0"
+torch_dtype = torch.bfloat16
+model_name = "parler-tts/parler-tts-mini-v1"
+
+# need to set padding max length
+max_length = 50
+
+# load model and tokenizer
+tokenizer = AutoTokenizer.from_pretrained(model_name) 
+model = ParlerTTSForConditionalGeneration.from_pretrained(
+    model_name,
+).to(torch_device, dtype=torch_dtype)
+
+sampling_rate = model.audio_encoder.config.sampling_rate
+frame_rate = model.audio_encoder.config.frame_rate
+
+def generate(text, description, play_steps_in_s=0.5):
+  play_steps = int(frame_rate * play_steps_in_s)
+  streamer = ParlerTTSStreamer(model, device=torch_device, play_steps=play_steps)
+  # tokenization
+  inputs = tokenizer(description, return_tensors="pt").to(torch_device)
+  prompt = tokenizer(text, return_tensors="pt").to(torch_device)
+  # create generation kwargs
+  generation_kwargs = dict(
+    input_ids=inputs.input_ids,
+    prompt_input_ids=prompt.input_ids,
+    attention_mask=inputs.attention_mask,
+    prompt_attention_mask=prompt.attention_mask,
+    streamer=streamer,
+    do_sample=True,
+    temperature=1.0,
+    min_new_tokens=10,
+  )
+  # initialize Thread
+  thread = Thread(target=model.generate, kwargs=generation_kwargs)
+  thread.start()
+  # iterate over chunks of audio
+  for new_audio in streamer:
+    if new_audio.shape[0] == 0:
+      break
+    print(f"Sample of length: {round(new_audio.shape[0] / sampling_rate, 4)} seconds")
+    yield sampling_rate, new_audio
+
+
+# now you can do
+text = "This is a test of the streamer class"
+description = "Jon's talking really fast."
+
+chunk_size_in_s = 0.5
+
+for (sampling_rate, audio_chunk) in generate(text, description, chunk_size_in_s):
+  # You can do everything that you need with the chunk now
+  # For example: stream it, save it, play it.
+  print(audio_chunk.shape) 
```

Comment:
Maybe we can link the streaming Gradio demo here to give users an example of how it's used in practice?

Reply:
One option would be to include a dropdown that shows a PyAudio example as well:

<details>

<summary> PyAudio Example </summary>

Show an example inspired by https://stackoverflow.com/a/32161871

</details>

Reply:
Unfortunately I can't test it easily on DGX ! (missing some utils)

Reply:
@sanchit-gandhi @ylacombe could you please give an example for playing the chunk, with pyaudio or something else, please ?